public class junit.framework.TestResult extends java.lang.Object
{
    protected java.util.Vector fErrors;
    protected java.util.Vector fFailures;
    protected java.util.Vector fListeners;
    protected int fRunTests;
    private boolean fStop;

    public void <init>()
    {
        junit.framework.TestResult $r0;
        java.util.Vector $r1;

        $r0 := @this: junit.framework.TestResult;

        specialinvoke $r0.<java.lang.Object: void <init>()>();

        $r1 = new java.util.Vector;

        specialinvoke $r1.<java.util.Vector: void <init>()>();

        $r0.<junit.framework.TestResult: java.util.Vector fFailures> = $r1;

        $r1 = new java.util.Vector;

        specialinvoke $r1.<java.util.Vector: void <init>()>();

        $r0.<junit.framework.TestResult: java.util.Vector fErrors> = $r1;

        $r1 = new java.util.Vector;

        specialinvoke $r1.<java.util.Vector: void <init>()>();

        $r0.<junit.framework.TestResult: java.util.Vector fListeners> = $r1;

        $r0.<junit.framework.TestResult: int fRunTests> = 0;

        $r0.<junit.framework.TestResult: boolean fStop> = 0;

        return;
    }

    private java.util.Vector cloneListeners()
    {
        junit.framework.TestResult $r0;
        java.lang.Object $r1;
        java.util.Vector $r2;
        java.lang.Throwable $r3;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r2 = $r0.<junit.framework.TestResult: java.util.Vector fListeners>;

        $r1 = virtualinvoke $r2.<java.util.Vector: java.lang.Object clone()>();

        $r2 = (java.util.Vector) $r1;

     label2:
        exitmonitor $r0;

        return $r2;

     label3:
        $r3 := @caughtexception;

        exitmonitor $r0;

        throw $r3;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public void addError(junit.framework.Test, java.lang.Throwable)
    {
        junit.framework.TestResult $r0;
        junit.framework.Test $r1;
        java.lang.Throwable $r2, $r8;
        junit.framework.TestFailure $r3;
        java.util.Vector $r4;
        java.util.Enumeration $r5;
        boolean $z0;
        java.lang.Object $r6;
        junit.framework.TestListener $r7;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.Test;

        $r2 := @parameter1: java.lang.Throwable;

        entermonitor $r0;

     label1:
        $r4 = $r0.<junit.framework.TestResult: java.util.Vector fErrors>;

        $r3 = new junit.framework.TestFailure;

        specialinvoke $r3.<junit.framework.TestFailure: void <init>(junit.framework.Test,java.lang.Throwable)>($r1, $r2);

        virtualinvoke $r4.<java.util.Vector: void addElement(java.lang.Object)>($r3);

        $r4 = specialinvoke $r0.<junit.framework.TestResult: java.util.Vector cloneListeners()>();

        $r5 = virtualinvoke $r4.<java.util.Vector: java.util.Enumeration elements()>();

     label2:
        $z0 = interfaceinvoke $r5.<java.util.Enumeration: boolean hasMoreElements()>();

        if $z0 == 0 goto label5;

        $r6 = interfaceinvoke $r5.<java.util.Enumeration: java.lang.Object nextElement()>();

        $r7 = (junit.framework.TestListener) $r6;

        interfaceinvoke $r7.<junit.framework.TestListener: void addError(junit.framework.Test,java.lang.Throwable)>($r1, $r2);

     label3:
        goto label2;

     label4:
        $r8 := @caughtexception;

        exitmonitor $r0;

        throw $r8;

     label5:
        exitmonitor $r0;

        return;

        catch java.lang.Throwable from label1 to label3 with label4;
    }

    public void addFailure(junit.framework.Test, junit.framework.AssertionFailedError)
    {
        junit.framework.TestResult $r0;
        junit.framework.Test $r1;
        junit.framework.AssertionFailedError $r2;
        junit.framework.TestFailure $r3;
        java.util.Vector $r4;
        java.util.Enumeration $r5;
        boolean $z0;
        java.lang.Object $r6;
        junit.framework.TestListener $r7;
        java.lang.Throwable $r8;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.Test;

        $r2 := @parameter1: junit.framework.AssertionFailedError;

        entermonitor $r0;

     label1:
        $r4 = $r0.<junit.framework.TestResult: java.util.Vector fFailures>;

        $r3 = new junit.framework.TestFailure;

        specialinvoke $r3.<junit.framework.TestFailure: void <init>(junit.framework.Test,java.lang.Throwable)>($r1, $r2);

        virtualinvoke $r4.<java.util.Vector: void addElement(java.lang.Object)>($r3);

        $r4 = specialinvoke $r0.<junit.framework.TestResult: java.util.Vector cloneListeners()>();

        $r5 = virtualinvoke $r4.<java.util.Vector: java.util.Enumeration elements()>();

     label2:
        $z0 = interfaceinvoke $r5.<java.util.Enumeration: boolean hasMoreElements()>();

        if $z0 == 0 goto label5;

        $r6 = interfaceinvoke $r5.<java.util.Enumeration: java.lang.Object nextElement()>();

        $r7 = (junit.framework.TestListener) $r6;

        interfaceinvoke $r7.<junit.framework.TestListener: void addFailure(junit.framework.Test,junit.framework.AssertionFailedError)>($r1, $r2);

     label3:
        goto label2;

     label4:
        $r8 := @caughtexception;

        exitmonitor $r0;

        throw $r8;

     label5:
        exitmonitor $r0;

        return;

        catch java.lang.Throwable from label1 to label3 with label4;
    }

    public void addListener(junit.framework.TestListener)
    {
        junit.framework.TestResult $r0;
        junit.framework.TestListener $r1;
        java.util.Vector $r2;
        java.lang.Throwable $r3;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.TestListener;

        entermonitor $r0;

     label1:
        $r2 = $r0.<junit.framework.TestResult: java.util.Vector fListeners>;

        virtualinvoke $r2.<java.util.Vector: void addElement(java.lang.Object)>($r1);

     label2:
        exitmonitor $r0;

        return;

     label3:
        $r3 := @caughtexception;

        exitmonitor $r0;

        throw $r3;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public void endTest(junit.framework.Test)
    {
        junit.framework.Test $r0;
        junit.framework.TestResult $r1;
        java.util.Vector $r2;
        java.util.Enumeration $r3;
        boolean $z0;
        java.lang.Object $r4;
        junit.framework.TestListener $r5;

        $r1 := @this: junit.framework.TestResult;

        $r0 := @parameter0: junit.framework.Test;

        $r2 = specialinvoke $r1.<junit.framework.TestResult: java.util.Vector cloneListeners()>();

        $r3 = virtualinvoke $r2.<java.util.Vector: java.util.Enumeration elements()>();

     label1:
        $z0 = interfaceinvoke $r3.<java.util.Enumeration: boolean hasMoreElements()>();

        if $z0 == 0 goto label2;

        $r4 = interfaceinvoke $r3.<java.util.Enumeration: java.lang.Object nextElement()>();

        $r5 = (junit.framework.TestListener) $r4;

        interfaceinvoke $r5.<junit.framework.TestListener: void endTest(junit.framework.Test)>($r0);

        goto label1;

     label2:
        return;
    }

    public int errorCount()
    {
        junit.framework.TestResult $r0;
        int $i0;
        java.util.Vector $r1;
        java.lang.Throwable $r2;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r1 = $r0.<junit.framework.TestResult: java.util.Vector fErrors>;

        $i0 = virtualinvoke $r1.<java.util.Vector: int size()>();

     label2:
        exitmonitor $r0;

        return $i0;

     label3:
        $r2 := @caughtexception;

        exitmonitor $r0;

        throw $r2;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public java.util.Enumeration errors()
    {
        junit.framework.TestResult $r0;
        java.util.Enumeration $r1;
        java.util.Vector $r2;
        java.lang.Throwable $r3;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r2 = $r0.<junit.framework.TestResult: java.util.Vector fErrors>;

        $r1 = virtualinvoke $r2.<java.util.Vector: java.util.Enumeration elements()>();

     label2:
        exitmonitor $r0;

        return $r1;

     label3:
        $r3 := @caughtexception;

        exitmonitor $r0;

        throw $r3;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public int failureCount()
    {
        junit.framework.TestResult $r0;
        int $i0;
        java.util.Vector $r1;
        java.lang.Throwable $r2;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r1 = $r0.<junit.framework.TestResult: java.util.Vector fFailures>;

        $i0 = virtualinvoke $r1.<java.util.Vector: int size()>();

     label2:
        exitmonitor $r0;

        return $i0;

     label3:
        $r2 := @caughtexception;

        exitmonitor $r0;

        throw $r2;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public java.util.Enumeration failures()
    {
        junit.framework.TestResult $r0;
        java.util.Enumeration $r1;
        java.util.Vector $r2;
        java.lang.Throwable $r3;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r2 = $r0.<junit.framework.TestResult: java.util.Vector fFailures>;

        $r1 = virtualinvoke $r2.<java.util.Vector: java.util.Enumeration elements()>();

     label2:
        exitmonitor $r0;

        return $r1;

     label3:
        $r3 := @caughtexception;

        exitmonitor $r0;

        throw $r3;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public void removeListener(junit.framework.TestListener)
    {
        junit.framework.TestResult $r0;
        junit.framework.TestListener $r1;
        java.util.Vector $r2;
        java.lang.Throwable $r3;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.TestListener;

        entermonitor $r0;

     label1:
        $r2 = $r0.<junit.framework.TestResult: java.util.Vector fListeners>;

        virtualinvoke $r2.<java.util.Vector: boolean removeElement(java.lang.Object)>($r1);

     label2:
        exitmonitor $r0;

        return;

     label3:
        $r3 := @caughtexception;

        exitmonitor $r0;

        throw $r3;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    protected void run(junit.framework.TestCase)
    {
        junit.framework.TestResult $r0;
        junit.framework.TestCase $r1;
        junit.framework.TestResult$1 $r2;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.TestCase;

        virtualinvoke $r0.<junit.framework.TestResult: void startTest(junit.framework.Test)>($r1);

        $r2 = new junit.framework.TestResult$1;

        specialinvoke $r2.<junit.framework.TestResult$1: void <init>(junit.framework.TestResult,junit.framework.TestCase)>($r0, $r1);

        virtualinvoke $r0.<junit.framework.TestResult: void runProtected(junit.framework.Test,junit.framework.Protectable)>($r1, $r2);

        virtualinvoke $r0.<junit.framework.TestResult: void endTest(junit.framework.Test)>($r1);

        return;
    }

    public int runCount()
    {
        junit.framework.TestResult $r0;
        java.lang.Throwable $r1;
        int i0;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        i0 = $r0.<junit.framework.TestResult: int fRunTests>;

     label2:
        exitmonitor $r0;

        return i0;

     label3:
        $r1 := @caughtexception;

        exitmonitor $r0;

        throw $r1;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public void runProtected(junit.framework.Test, junit.framework.Protectable)
    {
        junit.framework.TestResult $r0;
        junit.framework.Test $r1;
        junit.framework.Protectable $r2;
        junit.framework.AssertionFailedError $r3;
        java.lang.ThreadDeath $r4;
        java.lang.Throwable $r5;

        $r0 := @this: junit.framework.TestResult;

        $r1 := @parameter0: junit.framework.Test;

        $r2 := @parameter1: junit.framework.Protectable;

     label1:
        interfaceinvoke $r2.<junit.framework.Protectable: void protect()>();

     label2:
        return;

     label3:
        $r3 := @caughtexception;

        virtualinvoke $r0.<junit.framework.TestResult: void addFailure(junit.framework.Test,junit.framework.AssertionFailedError)>($r1, $r3);

        return;

     label4:
        $r4 := @caughtexception;

        throw $r4;

     label5:
        $r5 := @caughtexception;

        virtualinvoke $r0.<junit.framework.TestResult: void addError(junit.framework.Test,java.lang.Throwable)>($r1, $r5);

        return;

        catch junit.framework.AssertionFailedError from label1 to label2 with label3;
        catch java.lang.ThreadDeath from label1 to label2 with label4;
        catch java.lang.Throwable from label1 to label2 with label5;
    }

    public boolean shouldStop()
    {
        junit.framework.TestResult $r0;
        java.lang.Throwable $r1;
        boolean z0;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        z0 = $r0.<junit.framework.TestResult: boolean fStop>;

     label2:
        exitmonitor $r0;

        return z0;

     label3:
        $r1 := @caughtexception;

        exitmonitor $r0;

        throw $r1;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public void startTest(junit.framework.Test)
    {
        junit.framework.Test $r0;
        junit.framework.TestResult $r1;
        int $i0, $i1;
        java.util.Vector $r2;
        java.util.Enumeration $r3;
        boolean $z0;
        java.lang.Object $r4;
        junit.framework.TestListener $r5;
        java.lang.Throwable $r6;

        $r1 := @this: junit.framework.TestResult;

        $r0 := @parameter0: junit.framework.Test;

        $i0 = interfaceinvoke $r0.<junit.framework.Test: int countTestCases()>();

        entermonitor $r1;

     label1:
        $i1 = $r1.<junit.framework.TestResult: int fRunTests>;

        $i0 = $i1 + $i0;

        $r1.<junit.framework.TestResult: int fRunTests> = $i0;

        exitmonitor $r1;

     label2:
        $r2 = specialinvoke $r1.<junit.framework.TestResult: java.util.Vector cloneListeners()>();

        $r3 = virtualinvoke $r2.<java.util.Vector: java.util.Enumeration elements()>();

     label3:
        $z0 = interfaceinvoke $r3.<java.util.Enumeration: boolean hasMoreElements()>();

        if $z0 == 0 goto label7;

        $r4 = interfaceinvoke $r3.<java.util.Enumeration: java.lang.Object nextElement()>();

        $r5 = (junit.framework.TestListener) $r4;

        interfaceinvoke $r5.<junit.framework.TestListener: void startTest(junit.framework.Test)>($r0);

        goto label3;

     label4:
        $r6 := @caughtexception;

     label5:
        exitmonitor $r1;

     label6:
        throw $r6;

     label7:
        return;

        catch java.lang.Throwable from label1 to label2 with label4;
        catch java.lang.Throwable from label5 to label6 with label4;
    }

    public void stop()
    {
        junit.framework.TestResult $r0;
        java.lang.Throwable $r1;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $r0.<junit.framework.TestResult: boolean fStop> = 1;

     label2:
        exitmonitor $r0;

        return;

     label3:
        $r1 := @caughtexception;

        exitmonitor $r0;

        throw $r1;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public boolean wasSuccessful()
    {
        junit.framework.TestResult $r0;
        int $i0;
        boolean $z0;
        java.lang.Throwable $r1;

        $r0 := @this: junit.framework.TestResult;

        entermonitor $r0;

     label1:
        $i0 = virtualinvoke $r0.<junit.framework.TestResult: int failureCount()>();

        if $i0 != 0 goto label4;

        $i0 = virtualinvoke $r0.<junit.framework.TestResult: int errorCount()>();

     label2:
        if $i0 != 0 goto label4;

        $z0 = 1;

     label3:
        exitmonitor $r0;

        return $z0;

     label4:
        $z0 = 0;

        goto label3;

     label5:
        $r1 := @caughtexception;

        exitmonitor $r0;

        throw $r1;

        catch java.lang.Throwable from label1 to label2 with label5;
    }
}
